### ISSUE TITLE
Differentiating an extremely complicated weightless transform

### ISSUE THREAD

**Issue by OverLordGoldDragon** [tensorflow/tensorflow#36451]
I'm considering inserting a weightless transform between CNN layers in a neural network. The transform is very complicated - details below. I'm aware that custom transformations are supported via e.g. Keras'  `Lambda` layer  - however, I don't know if TF is capable of handling something of this complexity. The code I have needs to be converted for compatibility, but I rather it not be a wasted effort.

Can TensorFlow's autodifferentiation handle [`synsq_cwt_fwd`](https://github.com/OverLordGoldDragon/ssqueezepy/blob/master/ssqueezepy/synsq_cwt.py#L12)?

<hr>

**Continuous Synchrosqueezing Transform** - full dev-stage code in the [ssqueezepy](https://github.com/OverLordGoldDragon/ssqueezepy) repository. Highlights (simplified):

```python
x = np.random.randn(2000)
len(Wx.squeeze().shape) == 2

psihfn = lambda w: np.exp(2 * PI * 1j * .1 * w) * (
    np.abs(w - 1) < .999) * np.exp(-1. / (1 - ((w - 1) * (np.abs(w - 1) < .999)) ** 2))

for i in range(200):
    psih = psihfn(5 * np.arange(500))
    Wx[i] = np.fft.ifftshift(np.fft.ifft(psih * np.fft.fft(x)))
```
```python
Wx.dtype == 'complex128'
u = np.unwrap(np.angle(Wx)).T
w = np.array([np.diff(u), u[-1] - u[0]]).T
```
 1. `psihfn` involves (a) complex exponential; (b) inverse asymptotic real exponential; (c) boolean mask
 2. `Wx` involves (a) `fft` of `ifftshift` of `ifft` of `psih * fft(x)`, (b) computed _iteratively_, but _not recurrently_ (future values don't depend on past)
 3. `w`  (second blob) involves (a) `Wx` computed in 1 & 2, a `complex128` 2D array; (b) applying `dif(unwrap(angle(Wx)).T)`

The complex components may be eliminated entirely, including in 1(a). The main subject of attention is perhaps the FFT nest; I don't know how numpy computes FFT, as it encapsulates an involved op. There's also a question of _performance_ - if it takes hours on a small array, it's not worth it.

**Comment by ccrusius** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
